src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs

```diff
@@ -79,6 +79,17 @@ public override void Execute()
         {
             base.Execute();
 
+            // One of SourceAddress or SourceIpGroup must be present
+            if ((SourceAddress == null) && (SourceIpGroup == null))
```

Comment:
if one of these two parameter is required, it need to be handled by parameter sets:

one with SourceAddress mandatory and the other with SourceIpGroup mandatory.

Reply:
@VeryEarly 
I have not added parameter sets before. How exactly does it work? Should I just create a class for Parameter sets, added 2 contant strings, one for each and then specific corresponding ones for each of these fields? I dont understand how would that ensure at least one of the 2 fields is always present.

Reply:
let's say you have two parameter sets only, parameters provided must be fulfill one of the parameter set.

Make SourceAddress  mandatory for one set and SourceIpGroup mandatory mandatory for the other.

Reply:
Done